encoders/firebase-decoders-json/src/test/java/com/google/firebase/decoders/json/JsonDataDecoderAnnotatedFieldHandlerTest.java

```diff
@@ -83,6 +84,19 @@ public int hashCode() {
     }
   }
 
+  private final static AnnotatedFieldHandler<Default> DEFAULT_HANDLER =
+      new AnnotatedFieldHandler<Default>() {
+        @Nullable
+        @Override
+        public <T> T apply(
+            @NonNull Default annotation, @Nullable T fieldDecodedResult, @NonNull Class<T> type) {
```

Comment:
Just ran some experiments to remove `T`,
changes
```
  <T> T apply(@NonNull U annotation, @Nullable T fieldDecodedResult, @NonNull Class<T> type);
```
to
```
  Object apply(@NonNull U annotation, @Nullable Object fieldDecodedResult, @NonNull Class<?> type);
```

which allows us to write lambda like the following:
```
  private final static AnnotatedFieldHandler<Default> DEFAULT_HANDLER =
          (an, res, type) -> {
    if (res == null) {
      if (type.equals(String.class)) {
        return an.value();
      }
    }
    return res;
  };
```
But, we still not be able to directly register `AnnotatedFieldHandler` as lambda, since It still needs the info about the type of the Annotation, in that case, it's the `U` in above `apply` method.

Reply:
it should be possible to use the lambda in the `register()` method as it has the annotation type as first argument, but not possible to do it in the case of `Map<? extends Annotation, Handler<?>>` - which is fine as it is an internal implementation detail and should not be used directly anywhere